src/orchestrator-kubernetes/src/lib.rs

```diff
@@ -489,12 +635,67 @@ impl NamespacedOrchestrator for NamespacedKubernetesOrchestrator {
                 }
             };
 
+            let disk_usage = match disk_usage {
+                Some(metrics) => {
+                    if !metrics.items.is_empty() {
+                        let disk_usage_str = &*metrics.items[0].value.0;
+                        match parse_k8s_quantity(disk_usage_str) {
+                            Ok(q) => match q.try_to_integer(0, true) {
+                                Some(i) => Some(i),
+                                None => {
+                                    tracing::error!("Disk usage value {q:?} out of range");
+                                    None
+                                }
+                            },
+                            Err(e) => {
+                                tracing::error!(
+                                    "Failed to parse disk usage value {disk_usage_str}: {e}"
+                                );
+                                None
+                            }
+                        }
+                    } else {
+                        None
+                    }
+                }
+                None => None,
+            };
+            let disk_capacity = match disk_capacity {
+                Some(metrics) => {
+                    if !metrics.items.is_empty() {
+                        let disk_capacity_str = &*metrics.items[0].value.0;
+                        match parse_k8s_quantity(disk_capacity_str) {
+                            Ok(q) => match q.try_to_integer(0, true) {
+                                Some(i) => Some(i),
+                                None => {
+                                    tracing::error!("Disk capacity value {q:?} out of range");
+                                    None
+                                }
+                            },
+                            Err(e) => {
+                                tracing::error!(
+                                    "Failed to parse disk capacity value {disk_capacity_str}: {e}"
+                                );
+                                None
+                            }
+                        }
+                    } else {
+                        None
+                    }
+                }
+                None => None,
+            };
```

Comment:
Seems a bit odd not to just work this into the `match` above, instead of doing a separate `match`?

Reply:
I can try to merge these a bit